nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CaptureChangeMySQL.java

```diff
@@ -0,0 +1,928 @@
+            .required(true)
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.HOSTNAME_PORT_LIST_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor DRIVER_NAME = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-driver-class")
+            .displayName("MySQL Driver Class Name")
+            .description("The class name of the MySQL database driver class")
+            .defaultValue("com.mysql.jdbc.Driver")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor DRIVER_LOCATION = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-driver-locations")
+            .displayName("MySQL Driver Location(s)")
+            .description("Comma-separated list of files/folders and/or URLs containing the MySQL driver JAR and its dependencies (if any). "
+                    + "For example '/var/tmp/mysql-connector-java-5.1.38-bin.jar'")
+            .defaultValue(null)
+            .required(false)
+            .addValidator(StandardValidators.createListValidator(true, true, StandardValidators.createURLorFileValidator()))
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor USERNAME = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-username")
+            .displayName("Username")
+            .description("Username to access the MySQL cluster")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor PASSWORD = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-password")
+            .displayName("Password")
+            .description("Password to access the MySQL cluster")
+            .required(false)
+            .sensitive(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor DIST_CACHE_CLIENT = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-dist-map-cache-client")
+            .displayName("Distributed Map Cache Client")
+            .description("Identifies a Distributed Map Cache Client controller service to be used for keeping information about the various tables, columns, etc. "
+                    + "needed by the processor. If a client is not specified, the generated events will not include column type or name information.")
+            .identifiesControllerService(DistributedMapCacheClient.class)
+            .required(false)
```

Comment:
I assume every DML statement has a preceding Table Map event. If so, even if there's no table and column names cached, it can be retrieved by asking to the DBMS, then we probably do not need to persist it through a distributed cache.

For the large number of tables and columns, adding a cache limit in terms of number of tables might help.